### PR TITLE
Nl extra home page scenarios

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: src/test/resources/features/Login.feature [hunk 1]
```diff
@@ -8,3 +8,26 @@ Feature: Login
...
+
+  Scenario: Navigate to TDR Home Page as a logged in user with correct credentials and navigate away
+    Given A logged out user
+    When the logged out user visits url
+    And the logged out user clicks the .govuk-button--start element
+    Then the logged out user should be at the auth page
+    And the logged out user enters valid credentials
+    And the logged out user clicks the [name='login'] element
+    Then the logged in user should be at the dashboard page
+    When the logged in user navigates to google
```
Comment (TomJKing) [inline src/test/resources/features/Login.feature]: Could you amend the wording of this step to: When the logged in user navigates to different url

File: src/test/scala/steps/Steps.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  var webDriver: WebDriver = _
+  var userId: String = ""
+
+  val configuration = ConfigFactory.load()
+  val baseUrl: String = configuration.getString("tdr.base.url")
+  val authUrl: String = configuration.getString("tdr.auth.url")
+  val userName: String = RandomUtility.randomString()
+  val password: String = RandomUtility.randomString(10)
+  val otherPageUrl: String = configuration.getString("redirect.base.url")
+  val seriesPageUrl: String = configuration.getString("series.base.url")
```
Comment (TomJKing) [inline src/test/scala/steps/Steps.scala]: seriesPageUrl not needed as just require the base url, and pass in the page name from the feature step to complete the URL

File: src/test/scala/steps/Steps.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  Given("^A logged out user") {
+    userId = KeycloakClient.createUser(userName, password)
+  }
+
+  When("^the logged out user visits url") {
+    webDriver.get(s"$baseUrl")
+  }
+
+  When("^the logged in user visits series page") {
+    webDriver.get(s"$seriesPageUrl") // hardcoded the series page url, maybe not best idea, change this?
```
Comment (TomJKing) [inline src/test/scala/steps/Steps.scala]: Should have a generic step here. 
When("^the logged in user navigates to the (.*) page) {
webdriver.get(s"$baseUrl/$page")
}

File: src/test/scala/steps/Steps.scala [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+class Steps extends ScalaDsl with EN with Matchers {
+  var webDriver: WebDriver = _
+  var userId: String = ""
+
+  val configuration = ConfigFactory.load()
+  val baseUrl: String = configuration.getString("tdr.base.url")
+  val authUrl: String = configuration.getString("tdr.auth.url")
+  val userName: String = RandomUtility.randomString()
+  val password: String = RandomUtility.randomString(10)
+  val otherPageUrl: String = configuration.getString("redirect.base.url")
```
Comment (TomJKing) [inline src/test/scala/steps/Steps.scala]: Change to nonTdrPageUrl

File: src/test/resources/features/Login.feature (hunk 1 above)
Comment (techncl) [reply src/test/resources/features/Login.feature]: Resolved

File: src/test/resources/features/Login.feature [hunk 5]
```diff
@@ -8,3 +8,26 @@ Feature: Login
     And the logged out user enters valid credentials
     And the logged out user clicks the [name='login'] element
     Then the logged in user should be at the dashboard page
+
+  Scenario: Navigate to TDR Home Page as a logged out user with incorrect credentials
+    Given A logged out user
+    When the logged out user visits url
```
[the remark this answers is not in the document]
Comment (techncl) [reply src/test/resources/features/Login.feature]: Resolved

File: src/test/resources/features/Login.feature [hunk 6]
```diff
@@ -8,3 +8,26 @@ Feature: Login
...
+    And the logged out user clicks the .govuk-button--start element
+    Then the logged out user should be at the auth page
+    And the logged out user enters invalid credentials
+    And the logged out user clicks the [name='login'] element
+    Then the logged out user will remain on the auth page
+    And the user will see an error message
+
+  Scenario: Navigate to TDR Home Page as a logged in user with correct credentials and navigate away
+    Given A logged out user
+    When the logged out user visits url
```
[the remark this answers is not in the document]
Comment (techncl) [reply src/test/resources/features/Login.feature]: Resolved

File: application.conf [hunk 7]
```diff
@@ -1,2 +1,4 @@
 tdr.base.url="http://localhost:9000"
-tdr.auth.url="http://localhost:8080"
\ No newline at end of file
+tdr.auth.url="http://localhost:8080"
+redirect.base.url="https://www.google.co.uk/"
+series.base.url="http://localhost:9000/series"
```
[the remark this answers is not in the document]
Comment (techncl) [reply application.conf]: Resolved

File: src/test/scala/steps/Steps.scala (hunk 2 above)
Comment (techncl) [reply src/test/scala/steps/Steps.scala]: Resolved

File: src/test/scala/steps/Steps.scala (hunk 4 above)
Comment (techncl) [reply src/test/scala/steps/Steps.scala]: Resolved

File: src/test/scala/steps/Steps.scala (hunk 3 above)
Comment (techncl) [reply src/test/scala/steps/Steps.scala]: Resolved

File: src/test/scala/helpers/steps/StepsUtility.scala [hunk 8]
```diff
@@ -8,7 +8,7 @@ object StepsUtility {
     val chromeOptions: ChromeOptions = new ChromeOptions
     val chromeDriverLocation = System.getenv("CHROME_DRIVER")
 
-    chromeOptions.setHeadless(true)
+    chromeOptions.setHeadless(false)
```
[the remark this answers is not in the document]
Comment (techncl) [reply src/test/scala/helpers/steps/StepsUtility.scala]: Resolved

File: src/test/scala/steps/Steps.scala [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    StepsUtility.userLogin(webDriver, userName, password)
+  }
+
+  When("^the logged out user navigates to TDR Home Page") {
+    webDriver.get(s"$baseUrl")
+  }
+
+  When("^the logged in user navigates to the (.*) page") {
+    page: String =>
+      webDriver.get(s"$baseUrl/$page") // hardcoded the series page url, maybe not best idea, change this?
```
Comment (TomJKing) [inline src/test/scala/steps/Steps.scala]: Remove unnecessary comment

Comment (techncl) [reply src/test/scala/steps/Steps.scala]: Resolved

**Pull request merged**